### PR TITLE
Fix warning array_key_exists in deserialization.

### PR DESCRIPTION
**Pull request by leonnleite**
This problem has reported in PR #110 and #138 ,

but they didn't tests. So I took the liberty of repeating the PR, now I did the tests.

### CODE REVIEW

**Comment by brunonm** [comment]
+1

**Comment by piotrantosik** [comment]
:+1:

File: src/JMS/Serializer/GenericDeserializationVisitor.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ public function visitProperty(PropertyMetadata $metadata, $data, Context $contex
     {
         $name = $this->namingStrategy->translateName($metadata);
 
-        if (null === $data || ! array_key_exists($name, $data)) {
+        if (null === $data || ! is_array($data) || ! array_key_exists($name, $data)) {
```
Comment (schmittjoh) [inline src/JMS/Serializer/GenericDeserializationVisitor.php]: Shouldn't this be an exception?

**Comment by Gladhon** [comment]
+1

File: src/JMS/Serializer/GenericDeserializationVisitor.php (hunk 1 above)
Comment (leonnleite) [inline src/JMS/Serializer/GenericDeserializationVisitor.php]: @schmittjoh after nearly a year, I fix it ...

thank you @Gladhon, for notifying me

**Pull request merged**